### ISSUE TITLE
cbind when subsetting spots from samples

### ISSUE THREAD

**Issue by lmweber** [drighelli/SpatialExperiment#104]
I was working with some SPE objects and noticed `cbind` has some unexpected behavior when subsetting spots from samples.

Specifically, I have started with a SPE containing multiple samples and with 2 distinct types of tissue per sample, so I split the samples into two sub-objects containing tissue type 1 and tissue type 2 respectively. Then I perform quality control (QC) separately on each sub-object / tissue type and store various QC metrics in the sub-objects. However then `cbind` does not let me easily put the objects back together due to checks on `sample_id`.

First, I get the warning:
```
> spe <- cbind(spe_LC, spe_WM)
'sample_id's are duplicated across 'SpatialExperiment' objects to cbind; appending sample indices.
```

and then an error if I try to remove the appended sample indices:
```
> colData(spe)$sample_id <- gsub("\\.[0-9]$", "", colData(spe)$sample_id)
Error in .local(x, ..., value) : 
  Number of unique 'sample_id's is 18, but 9 were provided.
```

So I think some of our checks are too strict here. I'll leave this note here for now so we don't forget about it and we can discuss at some point. It is possible to work around this issue by not working with sub-objects, but I think the error above has the potential to create some issues for some users.

**Comment by nilseling** [comment]
Thanks @lmweber for posting this. I came here to raise the same issue. I find this behaviour a bit dangerous as it generates `sample_id` entries that do not match the initial assignment and will therefore break any relation between the `SpatialExperiment` object and e.g. external metadata. For inexperienced users it can be potentially confusing.

**Comment by drighelli** [comment]
Thanks @nilseling and @lmweber,

Maybe, we should reconsider our way of thinking on the sample_id(s).

I think that `SingleCellExperiment` leaves the choice of the sample names to the user...

**Comment by nilseling** [comment]
Yeah, the `SingleCellExperiment` container has no hard-coded entry for the sample ids. But I guess you need them for linking cells/spots to images. My recommendation is to handle the `sample_id` entry "as is" in the `colData` slot and to check that there are no duplicates in the `imgData` entry

**Comment by ndamond** [comment]
Thanks @lmweber, I have exactly the same issue and the new `sample_id` entries generated by `cbind` also prevent using downstream functions:

```
aggregateAcrossCells(spe,  ids = spe$patient_id, statistics = "mean")
Error in .local(x, ..., value) : 
  Number of unique 'sample_id's is 47, but 1 was provided.
```

Is there a "better" workaround than this? (i.e. that would allow to do the same thing without reverting to `SingleCellExperiment`)
```
sce <- as(spe, "SingleCellExperiment")
sce$sample_id <- sce$roi_id
aggregateAcrossCells(sce,  ids = sce$patient_id, statistics = "mean")

class: SingleCellExperiment 
dim: 44 47 
metadata(4): spillcomp stages cases subset
assays(1): counts
rownames(44): GHRL HBP ... GCG PPY
rowData names(10): index channel ... clustering shortName
colnames(47): 6044 6061 ... 8005 8008
colData names(32): CaseID Panel ... ids ncells
reducedDimNames(0):
mainExpName: NULL
altExpNames(0):
```

**Comment by estellad** [comment]
Hello there, 

Has this been addressed? I am unable to manually add `sample_id` to `colData()`. Tried to add a column with another name and then rename it to "sample_id", but it did not work. Unless I convert SPE to SCE. 

```
CARD_obj$sample_id <- "sample01"

Error in .local(x, ..., value) : 
  Number of unique 'sample_id's is 0, but 1 was provided.
```

Thank you!
Estella

**Comment by HelenaLC** [comment]
Sorry for having lost track here... Leaving `sample_id`s to the user sounds fine by me. Just wondering what do folks think would be the expected behavior for `imgData`-handling? Intuitively, I'd have implemented `cbind(spe, spe)` leaves `sample_id`s as is, but retains `imgData` of the 1st (i.e., where `sample_id` and `image_id` are identical for both)?
